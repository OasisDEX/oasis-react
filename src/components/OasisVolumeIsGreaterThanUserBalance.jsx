import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import styles from './OasisVolumeIsGreaterThanUserBalance.scss';
import CSSModules from 'react-css-modules/dist/index';


const propTypes = PropTypes && {};
const defaultProps = {
  offerMax: PropTypes.string.isRequired
};


class OasisVolumeIsGreaterThanUserBalance extends PureComponent {
  render() {
    return (
      <div styleName={styles.base}>
        Current volume is greater than offer maximum of <b>{this.props.offerMax}</b>
      </div>
    );
  }
}

OasisVolumeIsGreaterThanUserBalance.displayName = 'OasisVolumeIsGreaterThanUserBalance';
OasisVolumeIsGreaterThanUserBalance.propTypes = propTypes;
OasisVolumeIsGreaterThanUserBalance.defaultProps = defaultProps;
export default CSSModules(OasisVolumeIsGreaterThanUserBalance);
