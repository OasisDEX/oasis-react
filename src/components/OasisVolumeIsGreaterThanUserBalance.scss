.base {
  margin-top: 20px; padding: 10px;
  text-align: center;
  background-color: #000000;
  color: #fff;
}
