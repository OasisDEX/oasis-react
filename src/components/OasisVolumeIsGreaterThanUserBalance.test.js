/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisVolumeIsGreaterThanUserBalance from './OasisVolumeIsGreaterThanUserBalance';


describe('(Component) OasisVolumeIsGreaterThanUserBalance', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <OasisVolumeIsGreaterThanUserBalance {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
