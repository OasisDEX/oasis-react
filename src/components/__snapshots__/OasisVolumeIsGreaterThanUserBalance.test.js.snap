// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisVolumeIsGreaterThanUserBalance should render 1`] = `
<div>
  Current volume is greater than offer maximum of 
  <b />
</div>
`;
